controllers/useraccount/useraccount_controller.go

```diff
@@ -484,6 +500,93 @@ func (r *Reconciler) deleteUser(ctx context.Context, userAcc *toolchainv1alpha1.
 	return true, nil
 }
 
+// Returns `true` if the associated resources (configMap, role and role-binding) created for a user by console is deleted, `false` otherwise with the underlying error.
+// This function only looks for these resources in the namespace - openshift-console-user-settings -
+func (r *Reconciler) deleteUserResources(ctx context.Context, userID string) (bool, error) {
+
+	// new user will have a label introduced which can help retrieve resources, otherwise older ones will have to rely on the name of the resource being of type `user-settings-<UID>`
+	// delete ConfigMap, Role and RoleBinding
+	if deleted, err := r.deleteConfigMap(ctx, userID); err != nil || deleted {
+		return deleted, err
+	}
+	if deleted, err := r.deleteRole(ctx, userID); err != nil || deleted {
+		return deleted, err
+	}
+	if deleted, err := r.deleteRoleBinding(ctx, userID); err != nil || deleted {
+		return deleted, err
+	}
+	return false, nil
```

Comment:
Same concern as above. We do not watch CMs, RoleBindings, and roles here. We can't delete them one by one. We either have to explicitly re-reconcile in this case or just delete them at once and do not stop the reconcile loop.
I think returning true/fails if any of the resources were deleted doesn't make much sense it this case. I would just return an error if something fails. Otherwise try to delete everything at the same loop.